### ISSUE TITLE
v4 (^4.0.0-beta.3): all hooks fail with a path error, but 0 exit code

### ISSUE THREAD

**Issue by xCykrix** [typicode/husky#596]
Node: v10.16.2
NPM: v6.12.0
Husky: 4.0.0-beta.3
Environment(s): command prompt, cygwin bash

All scripts return a 0 exit code with a path error specified, while still running any hooks that have a configuration. It would need to go up one more directory or remove the `/cygwin/` section for the path to be proper, as `../../../../` is the cygwin folder in command prompt and bash alike.

```
xCykrix@samuel-laptop ~/projects/whitelist-engine
$ HUSKY_DEBUG=1 git checkout
husky:debug husky v4.0.0-beta.3 (created at 10/14/2019, 8:08:22 PM)
husky:debug post-checkout hook started
husky:debug Current working directory is /home/xCykrix/projects/whitelist-engine
husky:debug post-checkout hook started
.git/hooks/post-checkout: line 66: cd: ../../../../cygwin/home/xCykrix/projects/whitelist-engine: No such file or directory
husky:debug npx --no-install husky-run exited with 0 exit code
```
```
xCykrix@samuel-laptop ~/projects/whitelist-engine
$ HUSKY_DEBUG=1 git commit
husky:debug husky v4.0.0-beta.3 (created at 10/14/2019, 8:08:22 PM)
husky:debug pre-commit hook started
husky:debug Current working directory is /home/xCykrix/projects/whitelist-engine
husky:debug pre-commit hook started
.git/hooks/pre-commit: line 66: cd: ../../../../cygwin/home/xCykrix/projects/whitelist-engine: No such file or directory
husky > pre-commit (node v10.16.2)

> project_name@1.0.0 devops.eslint C:\cygwin\home\xCykrix\projects\whitelist-engine
> eslint . --config=./.eslintrc

husky:debug npx --no-install husky-run exited with 0 exit code
On branch master
nothing to commit, working tree clean
```

If you have any other questions, feel free to ask.

**Comment by xCykrix** [comment]
All the currently installed hooks from using the beta, if they are of assistance.
[hooks.zip](https://github.com/typicode/husky/files/3726995/hooks.zip)

**Comment by typicode** [comment]
Thanks! I'll have a look.

**Comment by typicode** [comment]
Could you also post the output of `HUSKY_DEBUG=1 npm install husky --save-dev`?

**Comment by xCykrix** [comment]
Sorry for the delay! I had a really busy week.

Here is the output
```
xCykrix@samuel-laptop ~/projects/Discord
$ HUSKY_DEBUG=1 npm install husky@next --save-dev

> husky@4.0.0-beta.3 preuninstall C:\cygwin\home\xCykrix\projects\Discord\node_modules\husky
> node husky uninstall

husky:debug Current working directory is C:\cygwin\home\xCykrix\projects\Discord\node_modules\husky
husky:debug INIT_CWD environment variable is set to C:\cygwin\home\xCykrix\projects\Discord
husky > Uninstalling git hooks
husky:debug Git rev-parse command returned:
husky:debug   --show-top-level: /home/xCykrix/projects/Discord
husky:debug   --git-common-dir: ../../.git
husky > Done

> husky@4.0.0-beta.3 install C:\cygwin\home\xCykrix\projects\Discord\node_modules\husky
> node husky install

husky:debug Current working directory is C:\cygwin\home\xCykrix\projects\Discord\node_modules\husky
husky:debug INIT_CWD environment variable is set to C:\cygwin\home\xCykrix\projects\Discord
husky > Setting up git hooks
husky:debug Git rev-parse command returned:
husky:debug   --show-top-level: /home/xCykrix/projects/Discord
husky:debug   --git-common-dir: ../../.git
husky:debug Package manager: npm
husky:debug Installing hooks in ..\..\.git\hooks
husky > Done

> husky@4.0.0-beta.3 postinstall C:\cygwin\home\xCykrix\projects\Discord\node_modules\husky
> opencollective-postinstall || exit 0

Thank you for using husky!
If you rely on this package, please consider supporting our open collective:
> https://opencollective.com/husky/donate

npm WARN discord.js@11.5.1 requires a peer of @discordjs/uws@^10.149.0 but none is installed. You must install peer dependencies yourself.

+ husky@4.0.0-beta.3
updated 1 package and audited 928 packages in 9.298s
found 0 vulnerabilities
```

**Comment by typicode** [comment]
No worries, thanks for the logs. Could you try with the recently published `v4.0.0-beta.5` (it can be installed with `husky@next` tag)?

**Comment by xCykrix** [comment]
`v4.0.0-beta.5` has solved this issue, thank you @typicode. :slightly_smiling_face:
